docs/contributing/999-rfcs/2024-03-31-wing-secrets-cli.md

```diff
@@ -0,0 +1,54 @@
+---
+title: "#6105 Wing Secrets CLI"
+description: Creating Wing secrets from the CLI
+---
+
+# Wing Secrets CLI
+- **Author(s)**: @hasanaburayyan
+- **Submission Date**: 2024-03-31
+- **Stage**: Draft
+
+Creating secrets through the Wing CLI.
+
+## Background
+
+TODO
+
+## Sketch
+
+Running the blow command will result in an interactive prompt where you can input the Slack credentials:
+
+```bash
+wing secrets main.w --create
```

Comment:
What would happen if you compiled `main.w` without running `wing secrets main.w --create` first? Are there errors raised because the secrets are missing? (do the errors tell you the names of the missing secrets?)

Reply:
Kind of feels like it would make sense to ask for these secrets interactively when running in the simulator without secrets in the `.env` file, no?

Reply:
> What would happen if you compiled main.w without running wing secrets main.w --create first? 

The same thing as today, the code compiles fine, and a runtime error occurs. In my opinion that might be the better DX, since I imagine if Im just writing some code that expects a secret, I might not want an error or warning on each compile as it would get noisey.

Reply:
> Kind of feels like it would make sense to ask for these secrets interactively when running in the simulator without secrets in the .env file, no?

Im thinking to scope this mvp to just secret creation, and not retrieval for now. I kinda worry that these checking for secret values on compile might get noisey/annoying.